pcap/search.go

```diff
@@ -136,22 +136,63 @@ func genICMPFilter(src, dst net.IP) PacketFilter {
 	}
 }
 
-// XXX currently assumes legacy pcap is produced by the input reader
 // XXX need to handle searching over multiple pcap files
 func (s *Search) Run(ctx context.Context, w io.Writer, r pcapio.Reader) error {
+	reader := s.Reader(r)
+	_, err := reader.WriteToWithContext(ctx, w)
```

Comment:
Maybe shorten to `_, err := s.Reader(r).WriteToWithContext(ctx, w)`?

Reply:
Sure.